python/cuml/neighbors/nearest_neighbors.pyx

```diff
@@ -421,10 +432,13 @@ class NearestNeighbors(Base):
 
         if self._should_downcast:
             warnings.warn("Parameter should_downcast is deprecated, use "
-                          "convert_dtype in fit, fit_transform and transform "
+                          "convert_dtype in fit and kneighbors"
                           " methods instead. ")
             convert_dtype = True
 
+        if self._should_downcast is None:
+            self._should_downcast = True
```

Comment:
I think self._should_downcast should be set in the initializer, not in this method.